src/main/scala/com/cognite/spark/datasource/FilesRelation.scala

```diff
@@ -37,13 +20,15 @@ class FilesRelation(config: RelationConfig)(val sqlContext: SQLContext)
     client.files.updateFromRead(files) *> IO.unit
   }
 
-  override def clientToResource(client: GenericClient[IO, Nothing]): Files[IO] =
-    client.files
+  override def getStreams(filters: Array[Filter])(
+      client: GenericClient[IO, Nothing],
+      limit: Option[Long],
+      numPartitions: Int): Seq[fs2.Stream[IO, File]] =
+    Seq(
+      config.limit.map(client.files.listWithLimit(_)).getOrElse(client.files.list)
```

Comment:
You can leave out `(_)` I think, the newline after `Seq(` seems excessive but alright ;)

Reply:
Couldn't get rid of `(_)` it seems, apparently couldn't decide between `scala.Option.map` and `cats.Functor.Ops.map`. Agree one-liner is better, though.